src/code/middleware/firebase-imp.ts

```diff
@@ -114,6 +119,14 @@ export class FirebaseImp {
     this.pendingCallbacks=[];
   }
 
+  signOut() {
+    if (!this.isSignedOut || this.user) {
+      this.isSignedOut = true;
+      firebase.database().goOffline();
+      firebase.auth().signOut();
```

Comment:
I find the above surprising, but firebases API often surprises me.  

If this works and seems consistent with their documentation, its good enough for me.  👍

Reply:
I agree that the firebase API isn't always intuitive and the documentation isn't always as helpful as one might like. `signOut()` isn't sufficient by itself to remove the presence, and `goOffline()` doesn't say anything about authentication, so I'm assuming we should probably do both. Spurred by your comment I did a bit of testing and the order doesn't seem to matter, so I reversed the order and added a couple of comments in hopes of making things a bit clearer.